algebird-test/src/test/scala/com/twitter/algebird/Generators.scala

```diff
@@ -1,100 +0,0 @@
-/*
-Copyright 2013 Twitter, Inc.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-package com.twitter.algebird
-
-import org.scalacheck.Arbitrary
-import org.scalacheck.Gen
-import org.scalacheck.Gen._
-import Arbitrary.arbitrary
-
-case class PosNum[T: Numeric](value: T)
-
-object PosNum {
-  implicit def arb[T: Numeric: Gen.Choose]: Arbitrary[PosNum[T]] =
-    Arbitrary(for { p <- Gen.posNum[T] } yield PosNum(p))
-}
-
-case class NonEmptyVector[T](items: Vector[T]) {
-  def sorted(implicit ev: Ordering[T]): Vector[T] = items.sorted
-}
-
-object NonEmptyVector {
-  implicit def arb[T: Ordering: Arbitrary]: Arbitrary[NonEmptyVector[T]] =
-    Arbitrary(for {
-      l <- Gen.nonEmptyListOf(arbitrary[T])
-    } yield NonEmptyVector[T](l.toVector))
-}
-
-/**
- * Generators useful in testing Interval
- */
-object Generators {
-  // The new scalacheck oneOf results in diverging implicits here
-  // should follow up an investigation?
-  def oneOf[T](g1: Gen[T], g2: Gen[T], gs: Gen[T]*) = for {
```

Comment:
I managed to get rid of this diverging implicit expansion by adding a `[T]` parameter onto the `intervalArb` sub-generators.